lib/active_merchant/billing/gateways/visanet_peru.rb

```diff
@@ -174,23 +199,30 @@ def success_from(response)
         response["errorCode"] == 0
       end
 
-      def message_from(response)
+      def message_from(response, options, action)
         if empty?(response["errorMessage"]) || response["errorMessage"] == "[ ]"
-          response["data"]["DSC_COD_ACCION"]
+          "#{response["data"]["DSC_COD_ACCION"]}, #{options[:error_message]}" if action == 'refund'
+          response['data']['DSC_COD_ACCION']
         else
-          response["errorMessage"]
+          if action == 'refund'
+            message = "#{response["errorMessage"]}, #{options[:error_message]}"
+            options[:error_message] = response["errorMessage"]
```

Comment:
Is this line needed? `options` is not referenced or returned anywhere after this assignment.

Reply:
In this new configuration, message_from will be called twice (once for cancelDeposit and once for refundSingleTransaction). If it fails in the first call, I want to save the error message so I can show it in the second call. I wanted to capture both messages incase these calls failed for a reason other than the transaction wasn't settled.